File: app/src/main/java/com/studio1r/simpleblockexplorer/EOSApiService.java
package com.studio1r.simpleblockexplorer;


import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface EOSApiService {
    @GET("get_info")
    Observable<BlockInfo> getInfo();

    @POST("get_block")
    Observable<Block> getBlock(@Field("block_num_or_id") int blockId);
}

File: app/src/main/java/com/studio1r/simpleblockexplorer/EOSApiService.kt
package com.studio1r.simpleblockexplorer


import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface EOSApiService {
    @get:GET("get_info")
    val info: Observable<BlockInfo>

    @POST("get_block")
    fun getBlock(@Body() blockId: BlockInfoRequest): Observable<BlockEntity>
}
